akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -41,16 +42,38 @@ trait Projection[Envelope] {
   /**
    * Run the Projection.
    */
-  def run()(implicit systemProvider: ClassicActorSystemProvider): Unit
+  def run()(implicit systemProvider: ClassicActorSystemProvider): RunningProjection
```

Comment:
make those internal

Reply:
I will, that's super draft. I just wanted to validate the idea